### PR TITLE
Console-3733: Add support for Azure Workload Identity / Federated Identity based installs in OperatorHub

### PR DESCRIPTION
**Pull request by gallettilance**
This PR adds the following:

1. A warning is displayed in the item modal that the cluster is in WI/FI mode if the operator claims to supports it
2. A warning is displayed in the item subscription page that the cluster is in WI/FI mode if the operator claims to support it
3. If the cluster is in WI/FI mode and the operator claims support for it the the subscription page provides configuring 3 additional fields: client ID, tenant ID, and subscription ID (we are assuming that token path will be standardized / defaulted to `/var/run/secrets/openshift/serviceaccount/token` and region is not a required configuration)
4. Default subscription to manual for installs on WI/FI mode clusters for operators that support it

[Screencast from 2023-08-08 11-52-54.webm](https://github.com/openshift/console/assets/20077170/0f273eef-4fac-4974-aae2-35b4f55e635a)

### CODE REVIEW

**Comment by gallettilance** [comment]
cc @rhamilto @TheRealJon @jhadvig

**Comment by TheRealJon** [comment]
QE Approver
/assign @yanpzhan

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx:383]: Also, the infraFeatures check is not necessary for either of these warnings. The `isAzureWIFCluster` check is a prerequisite for the shortLivedTokenAuth infra feature to be included. Seen [here](https://github.com/openshift/console/pull/13082/files#diff-95817f2a930d7a83d3d022a955e9ca9fcd2bf275ccc5e6458378798df35dc017R189)

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by TheRealJon** [comment]
The comments I made are just nits. You can address them or not. Just let me know if you make changes and I'll review again.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx
Comment (gallettilance) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx:383]: @TheRealJon the warnings are mutually exclusive - an operator can support both AWS and Azure but the platform will only be one or the other.

Comment (gallettilance) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx:383]: Ah good catch on that duplication / redundancy

Comment (gallettilance) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx:383]: `shortLivedTokenAuth` is set to true when the operator supports the platform-specific short lived tokenized auth and the cluster is in that mode. So if anything it's the `isAzureWIFCluster` or `isAWSSTSCluster` that isn't required (I think) but I would like to display a warning custom to the platform (and `shortLivedTokenAuth` doesn't tell me the platform type) so I need a way to detect the platform type which is why `isAzure...` and `isAWS...` were added.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (gallettilance) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:440]: not sure I understand why - I need to know the operator claims support for the platform's specific short term auth method

**Comment by rhamilto** [comment]
cc: @jcaianirh

**Comment by gallettilance** [comment]
@rhamilto @jcaianirh this is ready for another round of reviews (just rebased)

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-utils.ts
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-utils.ts:23]: You're still using this.

**Review by rhamilto — COMMENTED** [review]
Nice job.  A few comments.

**Comment by rhamilto** [comment]
Nice work, @gallettilance.

/lgtm

**Comment by gallettilance** [comment]
QE Approver
/assign @yanpzhan 

Docs Approver:
/assign @opayne1 

PX Approver:
/assign @RickJWagner

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (opayne1) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:276]:
```suggestion
  "This cluster is using Azure Workload Identity / Federated Identity to reach the cloud API. In order for this operator to take the actions it requires directly with the cloud API, you will need to provide the Client ID, Tenant ID, and Subscription ID during installation. Please see the operator description for more details.": "This cluster is using Azure Workload Identity / Federated Identity to reach the cloud API. In order for this operator to take the actions it requires directly with the cloud API, provide the Client ID, Tenant ID, and Subscription ID during installation. See the operator description for more details.",
```
I am not sure if "please" is part of the PF guidelines, but in the RH documentation we avoid using "Please".

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json
Comment (opayne1) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:324]:
```suggestion
  "This cluster is using Azure Workload Identity / Federated Identity to reach the cloud API. In order for this operator to take the actions it requires directly with the cloud API, provide the Client ID, Tenant ID, and Subscription ID during installation. Manual subscriptions are highly recommended as steps should be taken before upgrading to ensure the permissions required by the next version are properly accounted for in the role. See the Operator description for more details.": "This cluster is using Azure Workload Identity / Federated Identity to reach the cloud API. In order for this operator to take the actions it requires directly with the cloud API, provide the Client ID, Tenant ID, and Subscription ID during installation. Manual subscriptions are highly recommended as steps should be taken before upgrade to ensure that the permissions required by the next version are properly accounted for in the role. See the operator description for more details.",
```
Again, following the RH documentation guidelines here.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx
Comment (opayne1) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:796]: Same docs related comments here.

**Review by opayne1 — COMMENTED** [review]
A couple docs related comments for you to consider.

**Comment by gallettilance** [comment]
@opayne1 Addressed your comments - thank you!

**Comment by gallettilance** [comment]
@yanpzhan thanks for catching that! You were right there was an issue with the new infraFeatures that I've fixed and it should all be working now

**Comment by yanpzhan** [comment]
@gallettilance Thanks for your update, I could see the warning info on new cluster launched against the pr now.
But there is other issue, twhen click on input fields "Azure Client ID","Azure Tenant ID","Azure Subscription ID", they are set as "[object Object]" automatically, and could not be updated. If insist on clicking "Install" button, an error shows up:
```
Danger alert:An error occurred
Converting circular structure to JSON
--> starting at object with constructor 'HTMLInputElement'
| property '__reactFiber$abxqa1swcim' -> object with constructor 'Lu'
--- property 'stateNode' closes the circle
```
Screenshot from 2023-12-06 17-47-38
Screenshot from 2023-12-06 17-50-16
Could you pls help to check again?

**Comment by rhamilto** [comment]
> ```
> Danger alert:An error occurred
> Converting circular structure to JSON
> --> starting at object with constructor 'HTMLInputElement'
> | property '__reactFiber$abxqa1swcim' -> object with constructor 'Lu'
> --- property 'stateNode' closes the circle
> ```

Related bug: https://issues.redhat.com/browse/ocpbugs-24252

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx:205]: My memory on this is fuzzy as it has been awhile and this has changed a bit.  `featuresAnnotationsObjects` really only exists in order to override legacy labels.  Since `tokenAuth*`s are new, I would not include them here.  I would preserve the existing functionality where you push the value on to `infrastructureFeatures`.

**Comment by gallettilance** [comment]
/hold cancel
/label px-approved

Adding these labels since QE will not be able to add them before branch date and this needs to be in before then and extensive testing has been done prior to today.

**Comment by yanpzhan** [comment]
Retested on cluster launched against the pr, the warning info show correctly and operator could be installed successfully.
I have a question about where the info set for "Azure Client Id", "Azure Tenant Id". "Azure Subscription Id" fields are stored after the operator is installed.  Pasted the comment in [jira](https://issues.redhat.com/browse/CONSOLE-3733)
@gallettilance could you give me some clue about the question? Thanks very much.

**Comment by rhamilto** [comment]
[Backported to 4.15 as a bug](https://github.com/openshift/console/pull/13428).

**Pull request merged**
